packages/gatsby/src/utils/api-runner-node.js

```diff
@@ -80,13 +80,46 @@ const initAPICallTracing = parentSpan => {
   }
 }
 
+const deferredAction = type => (...args) => {
+  emitter.emit(`ENQUEUE_NODE_MUTATION`, {
+    type,
+    payload: args,
+  })
+}
+
+const NODE_MUTATION_ACTIONS = [
+  `createNode`,
+  `deleteNode`,
+  `deleteNodes`,
+  `touchNode`,
+  `createParentChildLink`,
+  `createNodeField`,
+]
+
+const deferActions = actions => {
+  // console.log(`actual actions`, actions)
+  const returnValue = NODE_MUTATION_ACTIONS.reduce((prev, next) => {
+    prev[next] = deferredAction(next)
+    return prev
+  }, actions)
+  // console.log(`returnValue`, returnValue)
+  return returnValue
+}
+
 const getLocalReporter = (activity, reporter) =>
   activity
     ? { ...reporter, panicOnBuild: activity.panicOnBuild.bind(activity) }
     : reporter
 
+const pluginNodeCache = new Map()
+
 const runAPI = (plugin, api, args, activity) => {
-  const gatsbyNode = require(`${plugin.resolve}/gatsby-node`)
+  let gatsbyNode = pluginNodeCache.get(plugin.name)
+  if (!gatsbyNode) {
+    gatsbyNode = require(`${plugin.resolve}/gatsby-node`)
+    pluginNodeCache.set(plugin.name, gatsbyNode)
+  }
+
```

Comment:
This is perf, and non-functional change?

Reply:
Yeah. Dynamic requires are expensive at scale